### DISCUSSION TITLE
Toolbar icon size issue with second screen

### DISCUSSION THREAD

**Discussion by jeannich** [valinet/ExplorerPatcher#3138] [Q&A]
Hi there,

Depending on whether I plug-in my second screen BEFORE or AFTER booting my laptop, I'm seeing different Toolbar icon sizes and it does not follow the option "Taskbar icon size" of ExplorerPatcher

This screen shot explains it all:
![image](https://github.com/valinet/ExplorerPatcher/assets/31692007/f55c0407-bfab-4366-912b-827508a77296)

I was expecting Toolbar icon size would follow taskbar icon size settings but maybe it's a different concept?
Do you think I should file a bug report, or ask for an enhancement ?

**Comment by jeannich** [comment]
After having done all those tests to prepare the image above I can condense results with:
Taskbar icon size option works well in all my tests.
Toolbar icons size though is not correctly handled.


Analysis try:
What I observe tends to show that the **Toolbar icon size is calculated once for all when the laptop boots up**.
In my case I have a bigger resolution on my laptop than on my 2nd screen.

So if I start my laptop WITHOUT my second screen plugged-in, Toolbar icon size is calculated with a high resolution size. This is correctly displayed on my laptop of course, but when I plugin my second screen, those high resolution icons are displayed bigger than they should.

I start my laptop WITH my second screen plugged-in: in this case the Toolbar icon size seems to be calculated with the second screen resolution which is low res. In this case, toolbar icons are correctly displayed on my second screen, BUT if I unplug the second screen, then the Toolbar icons appears tiny on my higher res laptop screen.


Maybe we can summarize with:
    * --> Toolbar icon sizes should be recalculated at each multiple display changes (that would be a bug fix request)
    * --> Toolbar icon sizes should follow the settings "Taskbar icon size" (this could be an enhancement request)

**Comment by pyrates999** [comment]
Please redo your tests without EP installed.

**Reply by jeannich** [reply]
There is no toolbar feature on Windows 11

**Reply by pyrates999** [reply]
Did you change it to the windows 10 taskbar after installing EP?

**Reply by jeannich** [reply]
Yes, in EP I am using "Windows 10" taskbar style

**Reply by pyrates999** [reply]
Not sure if this will be fixed due to windows 11 24h2 removing the windows 10 start menu and windows 10 taskbar code entirely.

**Reply by jeannich** [reply]
oh, really ? Wow, Microsoft is really pushing their docking bar approach.
Too bad.

OK thanks for checking
